### PR TITLE
Include record primary constructor props in nav-to when not backed by a corresponding field/prop.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/57873

### CODE REVIEW

File: src/Workspaces/CSharp/Portable/FindSymbols/CSharpDeclaredSymbolInfoFactoryService.cs [hunk 1]
```diff
@@ -349,6 +349,55 @@ private static void ProcessUsings(List<Dictionary<string, string>> aliasMaps, Sy
...
+            StringTable stringTable,
+            ArrayBuilder<DeclaredSymbolInfo> declaredSymbolInfos,
+            string containerDisplayName,
+            string fullyQualifiedContainerName,
+            CancellationToken cancellationToken)
+        {
+            // Add synthesized properties for record primary constructors that are not backed by an existing field
+            // or property.
+            if (memberDeclaration is RecordDeclarationSyntax { ParameterList: { } parameterList } recordDeclaration &&
+                parameterList.Parameters.Count > 0)
```
Comment (sharwell) [inline src/Workspaces/CSharp/Portable/FindSymbols/CSharpDeclaredSymbolInfoFactoryService.cs:364]:
```suggestion
            if (memberDeclaration is RecordDeclarationSyntax { ParameterList: { Parameters.Count: > 0 } parameterList } recordDeclaration)
```

File: src/Workspaces/Core/Portable/LanguageServices/DeclaredSymbolFactoryService/AbstractDeclaredSymbolInfoFactoryService.cs [hunk 2]
```diff
@@ -205,6 +208,15 @@ protected static void Intern(StringTable stringTable, ArrayBuilder<string> build
                         memberDeclaration, child, stringTable, rootNamespace, declaredSymbolInfos, aliases, extensionMethodInfo,
                         innerContainerDisplayName, innerFullyQualifiedContainerName, cancellationToken);
                 }
+
+                AddSynthesizedDeclaredSymbolInfos(
```
[the remark this answers is not in the document]
Comment (sharwell) [reply src/Workspaces/Core/Portable/LanguageServices/DeclaredSymbolFactoryService/AbstractDeclaredSymbolInfoFactoryService.cs:212]: ➡️ GetChildren return MemberDeclarationSyntax, but record parameters are ParameterSyntax

File: src/Workspaces/CSharp/Portable/FindSymbols/CSharpDeclaredSymbolInfoFactoryService.cs [hunk 3]
```diff
@@ -198,13 +198,13 @@ private static void ProcessUsings(List<Dictionary<string, string>> aliasMaps, Sy
...
                             SyntaxKind.ClassDeclaration => DeclaredSymbolInfoKind.Class,
-                            SyntaxKind.RecordDeclaration => DeclaredSymbolInfoKind.Record,
                             SyntaxKind.InterfaceDeclaration => DeclaredSymbolInfoKind.Interface,
                             SyntaxKind.StructDeclaration => DeclaredSymbolInfoKind.Struct,
+                            SyntaxKind.RecordDeclaration => DeclaredSymbolInfoKind.Record,
                             SyntaxKind.RecordStructDeclaration => DeclaredSymbolInfoKind.RecordStruct,
                             _ => throw ExceptionUtilities.UnexpectedValue(node.Kind()),
                         },
-                        GetAccessibility(container, typeDecl.Modifiers),
+                        GetAccessibility(container, node.Modifiers),
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Workspaces/CSharp/Portable/FindSymbols/CSharpDeclaredSymbolInfoFactoryService.cs:207]: no. sorry, accidental fallout of moving something to/fro.

Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/FindSymbols/CSharpDeclaredSymbolInfoFactoryService.cs:207]:
```suggestion
                        GetAccessibility(container, typeDecl.Modifiers),
```

**Pull request merged**